### ISSUE TITLE
tfw AirFloat is fucking awesome and works on iPhone 6 / iOS 8

### ISSUE THREAD

**Issue by 19h** [trenskow/AirFloat#43]
#### tl;dr

**_Thank you!**_ — AirFloat is working with **_iOS 8**_ on my **_iPhone 6**_!

![](http://data.sly.mn/image/000V1N0E0g0A/ks.gif)
#### Abstract

It was a bit complicated to get AirFloat running on the iPhone 6 due to plain crazy new security measures. I'm a _registered_ developer and here's how I did it.

I obtained the app, repackaged it with my code signature and threw it on my iPhone with Xcode.
##### Getting the deb

Since I am not using Cydia, I had to find out how Cydia repos work in order to obtain the .deb from the repo (_I was not! able by any means to compile AirFloat_ — using Xcode 6.1 beta).
1. There's a Release file with info about the repo. — http://cydia.myrepospace.com/dervic/Release
2. There's a Packages file on the repo (wasn't there, it was gzipped) — http://cydia.myrepospace.com/dervic/Packages.gz ; extract it.
3. tl;dr
4. http://www.myrepospace.com//debfiles/3/32175/airfloat_1.3.2.deb
5. Extract the .tar file — you'll get an Applications/../ folder relative to root on the iDevice. Open your terminal in the AirFloat folder inside it.

You'll get an .app file.
##### Building the base structure.

Base structure: 

**iTunesArtwork**
**iTunesMetadata.plist**
**Payload**

Build the iTunesArtwork and iTunesMetadata.plist file.. or just copy them from a different app. I used the meta data from another app I released into the AppStore, so that it directly correlates with my certificate information. Name the app AirFloat inside iTunesMetadata.plist and use a name derived from your organization (e.g. **_com.kenansulayman**_.AirFloat). Move the AirFloat.app file into the ./Payload folder of your base structure.
##### Create a new, empty project to run on the device

Name it accordingly to your iTunesMetadata.plist file.

![](http://f.cl.ly/items/21142g181X3Q2h0T1R39/Screen%20Shot%202014-09-20%20at%2021.33.04.png)

Should look alike:

![](http://f.cl.ly/items/2C0u1B042r0e1s2k100D/Screen%20Shot%202014-09-20%20at%2021.36.28.png)
##### In case you're running into an error named "Security" (that's all, yeah)

Locate the AirFloat app on your device, click it and "trust" your provision profile. Remove the app.
##### Code-sign your app

Open keychain, locate your certificate —

![](http://f.cl.ly/items/1s3V091B1m2G1e1c3e1K/Screen%20Shot%202014-09-20%20at%2021.40.59.png)

Change to the Payload folder in your base structure and code-sign your application —

```
$ codesign -f -s "iPhone Developer: Your Name (*****)" AirFloat.app/
AirFloat.app/: replacing existing signature
```
##### Package your app

Just select all three files and compress them; rename the **Archive.zip** to **airfloat.ipa**.
##### Open Xcode / Devices and move AirFloat into the Applications field

![](http://f.cl.ly/items/1J0T1d0a2a1S2b1a0u2D/Screen%20Shot%202014-09-20%20at%2021.46.14.png)

....

![](https://sly.mn/wt/ks.gif)
### Using it

![](http://f.cl.ly/items/0q2127180Z2y0M0T0H0Z/Screen%20Shot%202014-09-20%20at%2021.50.28.png)
Yosemite: Select the AirPlay instance you're going to stream to.

**Comment by MikePolinske** [comment]
I also would like a finished product.  I'm new to app development and this is more complicated than I can easily do.

**Comment by paulyoung** [comment]
@KenanSulayman it appears that you're missing some steps.

Namely:
* we create an Xcode project and then do nothing with it
* we create an .ipa but the devices screen in Xcode requires a .app

Could you (or maybe @euwars) fill in the blanks?

**Comment by KenanSulayman** [comment]
1. The "empty" project is simply to get your certificate trusted on the device — and it will only ask to accept if you're building something. I simply used a binary of AirFloat, thus I had to install the empty project to get my certificate trusted.
2. Are you using the device manager dialog for it? It should accept the IPA; and the only reason we're creating it is the signature with which the device can work since we trusted it in 1.

**Comment by paulyoung** [comment]
I actually just cloned the repo, built the project, and ran it on my device with Xcode 6.3 (6D570) and iOS 8.3 (12F70) on my iPhone 5.

Thanks anyway.

**Comment by Chris2007a** [comment]
So I've just got the new jail break for my iPhone 6 Plus running iOS 8.3 and i discovered this tweak on youtube and gave it a try and i love it. Unfortunately it hasn't been updated and once my phone turns off and i turn it back on it crashes had to remove via safe mode. i don't know the first thing about Xcode in fact I'm installing it now. anyway one of you can help me get it to work?
